setup_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+import re
+
+
+def update_pins(setup_args):
+    # Use requirements and constraints to set version pins
+    packages = set()
+    with open('./requirements.txt') as requirements:
```

Comment:
Do we need to do anything about working directories here? I tried a few things and nothing broke, so I assume not?

Reply:
Good question. I guess we can use `os.path.dirname(__file__)`.